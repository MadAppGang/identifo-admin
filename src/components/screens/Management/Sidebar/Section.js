import React from 'react';
import PropTypes from 'prop-types';
import { NavLink, withRouter } from 'react-router-dom';

const SidebarSection = (props) => {
  const { path, title, match, Icon } = props;
  const isActive = match.params.section === path.split('/')[2];

  return (
    <NavLink
      exact
      to={path}
      className="iap-management-sidebar__section"
      activeClassName="iap-management-sidebar__section--active"
    >
      <Icon active={isActive} />
      <span>{title}</span>
    </NavLink>
  );
};

SidebarSection.propTypes = {
  path: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      section: PropTypes.string,
    }),
  }).isRequired,
  Icon: PropTypes.func.isRequired,
};

export { SidebarSection };

export default withRouter(SidebarSection);
