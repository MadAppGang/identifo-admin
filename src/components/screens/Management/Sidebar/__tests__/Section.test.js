import { SidebarSection } from '../Section';

describe('<SidebarSection />', () => {
  test('renders as expected for not selected section', () => {
    const props = {
      path: '/management/section',
      title: 'Section',
      match: {
        params: {
          section: '',
        },
      },
    };

    expect(shallow(<SidebarSection {...props} />)).toMatchSnapshot();
  });

  test('renders as expected for selected section', () => {
    const props = {
      path: '/management/section',
      title: 'Section',
      match: {
        params: {
          section: 'section',
        },
      },
    };

    expect(shallow(<SidebarSection {...props} />)).toMatchSnapshot();
  });
});
