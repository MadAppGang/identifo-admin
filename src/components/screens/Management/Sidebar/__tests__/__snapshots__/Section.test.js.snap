// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarSection /> renders as expected for not selected section 1`] = `
<NavLink
  activeClassName="iap-management-sidebar__section--active"
  aria-current="page"
  className="iap-management-sidebar__section"
  exact={true}
  to="/management/section"
>
  <mockConstructor
    active={false}
  />
  <span>
    Section
  </span>
</NavLink>
`;

exports[`<SidebarSection /> renders as expected for selected section 1`] = `
<NavLink
  activeClassName="iap-management-sidebar__section--active"
  aria-current="page"
  className="iap-management-sidebar__section"
  exact={true}
  to="/management/section"
>
  <mockConstructor
    active={true}
  />
  <span>
    Section
  </span>
</NavLink>
`;
